src/vscode-dts/vscode.d.ts

```diff
@@ -7731,6 +7731,17 @@ declare module 'vscode' {
 		 * https://invisible-island.net/xterm/ctlseqs/ctlseqs.html
 		 */
 		readonly isInteractedWith: boolean;
+
+		/**
+		 * The shell type of the terminal that may change throughout terminal's change of state.
```

Comment:
> that may change throughout terminal's change of state.

I'd put this in a different paragraph. For the first sentence just have a quick summary, maybe something like:

>  The shell type of the terminal, for example `bash`, `zsh`, ...

Then in a second paragraph you can explain how the `shell` value changes in more detail. I'd like to understand when this is undefined for example and what can cause this to chance

Reply:
Thanks for this. I made some changes: [9d92af0](https://github.com/microsoft/vscode/pull/243274/commits/9d92af0f3d7230b7260c15d383f63af804bee6fa) combining feedback from here with the one above. Let me know what you think!